### PR TITLE
virt-api, VMI update admitter: Remove file-system reads

### LINKED ISSUE #11177: Optimize IsKubeVirtServiceAccount()
**Is your feature request related to a problem? Please describe**:
https://github.com/kubevirt/kubevirt/blob/f081b2bf56f629788e46359c8d2e008fbe7834a7/pkg/virt-api/webhooks/utils.go#L83
`IsKubeVirtServiceAccount` is called from:
- VMI mutating webhook
- VMI creation validating webhook
- VMI update validating webhook

Every time this function is called, it [reads](https://github.com/kubevirt/kubevirt/blob/17517e765488f3dfd0c4272781d56522dc6051f4/staging/src/kubevirt.io/client-go/util/util.go#L13) a static file containing the namespace in which the `virt-api` pod is running.

**Describe the solution you'd like**:
Refactor the code in a way that will read the namespace file only once.

**Describe alternatives you've considered**:

**Additional context**:

### PR DESCRIPTION
**Pull request by orelmisan**
### What this PR does
PR #12009 introduced a check to see whether the VMI update had originated from the virt-handler.

Remove the file-system read made by `webhooks.GetNamespace()`.
Additionally, remove a call to `IsKubeVirtServiceAccount` that internally performs a file-system read.

Base the logic on the KubeVirtServiceAccount set introduced by PR #11931.

Deprecate `IsKubeVirtServiceAccount`.

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Special notes for your reviewer
Partially addresses #11177.
After this PR, a single call to `IsKubeVirtServiceAccount` is still left.
Removing it requires additional refactoring.

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
None
```

### CODE REVIEW

**Review by jean-edouard — COMMENTED** [review]
/approve
/hold
Just one suggestion in-line, feel free to unhold.

File: pkg/virt-api/webhooks/validating-webhook/admitters/vmi-update-admitter.go [hunk 1]
```diff
@@ -450,3 +452,11 @@ func admitHotplugMemory(oldMemory, newMemory *v1.Memory) *admissionv1.AdmissionR
 
 	return nil
 }
+
+func hasRequestOriginatedFrom(component, requestUsername string, kubeVirtServiceAccounts map[string]struct{}) bool {
+	if _, isKubeVirtServiceAccount := kubeVirtServiceAccounts[requestUsername]; isKubeVirtServiceAccount {
+		return strings.HasSuffix(requestUsername, component)
```
[the remark this answers is not in the document]
Comment (fossedihelm) [reply pkg/virt-api/webhooks/validating-webhook/admitters/vmi-update-admitter.go:458]: Even better from my POV:
```
func (admitter *VMIUpdateAdmitter) hasRequestOriginatedFromVirtHandler(requestUsername string) bool {
	if _, isKubeVirtServiceAccount := admitter.kubeVirtServiceAccounts[requestUsername]; isKubeVirtServiceAccount {
		return strings.HasSuffix(requestUsername, components.HandlerServiceAccountName)
	}

	return false
}
```

**Review by fossedihelm — APPROVED** [review]
Looks good!
As future improvement:
Create a VMIAdmitter initializer and unexport the fields
```
type VMIUpdateAdmitter struct {
	clusterConfig           *virtconfig.ClusterConfig
	kubeVirtServiceAccounts map[string]struct{}
}

func NewVMIAdmitter(config *virtconfig.ClusterConfig, serviceAccounts map[string]struct{}) *VMIUpdateAdmitter {
	return &VMIUpdateAdmitter{
		clusterConfig:           config,
		kubeVirtServiceAccounts: serviceAccounts,
	}
}
```
```

**Comment by orelmisan** [comment]
Change: Addressed @jean-edouard and @fossedihelm comments.

File: pkg/virt-api/webhooks/validating-webhook/admitters/vmi-update-admitter.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (orelmisan) [reply pkg/virt-api/webhooks/validating-webhook/admitters/vmi-update-admitter.go]: Done.

**Comment by orelmisan** [comment]
> Looks good! As future improvement: Create a VMIAdmitter initializer and unexport the fields
> 
> ```
> type VMIUpdateAdmitter struct {
> 	clusterConfig           *virtconfig.ClusterConfig
> 	kubeVirtServiceAccounts map[string]struct{}
> }
> 
> func NewVMIAdmitter(config *virtconfig.ClusterConfig, serviceAccounts map[string]struct{}) *VMIUpdateAdmitter {
> 	return &VMIUpdateAdmitter{
> 		clusterConfig:           config,
> 		kubeVirtServiceAccounts: serviceAccounts,
> 	}
> }
> ```

Good idea, I'll create a follow-up.

**Pull request merged**